### ISSUE TITLE
Semantic colorization for quoted type annotations is inconsistent

### ISSUE THREAD

**Issue by rchiodo** [microsoft/pylance-release#5810]
![image](https://github.com/microsoft/pylance-release/assets/19672699/9eb23ea2-c858-437a-8faf-a7083d6a8340)

It seems to me that the "Box[T]" should be all one color. The root cause is no semantic value for the `[` in the type string.

**Comment by debonte** [comment]
That matches the behavior of unquoted generic types.

![image](https://github.com/microsoft/pylance-release/assets/12820898/a0c37227-4e86-499b-a199-40212976a3a5)

**Comment by rchiodo** [comment]
That seems incorrect to me too :)

**Comment by debonte** [comment]
I disagree. The type var should be a different color to make it clear that it is a distinct identifier that supports, for example, go to def. I'm indifferent about the bracket coloring.

**Comment by rchiodo** [comment]
Sorry not the TypeVar, the brackets. Especially in the string. It looks super weird.

**Comment by debonte** [comment]
I see. The brackets are colored as string characters. I agree, that's weird. I'm not convinced that the brackets should be the same color as the type name though.

**Comment by rchiodo** [comment]
Decision was to provide semantic tokens for the brackets. That should make the colors consistent amongst the different ways to show a generic type.

**Comment by heejaechang** [comment]
currently, according to "Developer: Inspect Editor Tokens and Scopes", all those `[]` are currently marked as various `*.punctuation.*` or `*.quoted.*` tokens/scopes from `textmate`

so, we can either fix it from `semantic tokenizer` override `textmate` token/scopes or we could fix `textmate` ?

**Comment by heejaechang** [comment]
Repro workspace for investigation:
https://github.com/heejaechang/pylance-repro/tree/main/pylance-issues/5810

**Comment by heejaechang** [comment]
This issue has been fixed in prerelease version 2026.3.100, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20263100-24-August-2026-PreRelease)
